platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -366,4 +424,67 @@ private int findJoystickDevice(int device_id) {
 
 		return -1;
 	}
+
+	private boolean handleMouseEvent(final MotionEvent event) {
+		switch (event.getActionMasked()) {
+			case MotionEvent.ACTION_HOVER_ENTER:
+			case MotionEvent.ACTION_HOVER_MOVE:
+			case MotionEvent.ACTION_HOVER_EXIT: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int type = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.hover(type, x, y);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_BUTTON_PRESS:
+			case MotionEvent.ACTION_BUTTON_RELEASE:
+			case MotionEvent.ACTION_MOVE: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int buttonsMask = InputUtils.fixButtonsStateMask(event.getButtonState());
+				final int action = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.touch(event.getSource(), action, 0, 1, new float[] { 0, x, y }, buttonsMask, 1);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_SCROLL: {
+				final float[] positionArray = new float[] { 0, event.getX(), event.getY() };
+				final int buttonsMaskFixed = InputUtils.fixButtonsStateMask(event.getButtonState());
+
+				if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_UP, event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) < 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_DOWN, -event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				}
+				if (event.getAxisValue(MotionEvent.AXIS_HSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_RIGHT, event.getAxisValue(MotionEvent.AXIS_HSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_HSCROLL) < 0.0f) {
```

Comment:
Same here, this should be a regular `else` instead of an `else if`.

Reply:
same as vscroll